python/kwiver/sprokit/adapters/adapter_data_set.cxx

```diff
@@ -128,6 +131,35 @@ py::object get_port_data_correct_type(ka::adapter_data_set &self, ::sprokit::pro
   throw py::type_error(msg);
 }
 
+template < typename T >
+py::object value_or(ka::adapter_data_set &self,
+                    ::sprokit::process::port_t const& port,
+                    py::object value_if_missing=py::none())
+{
+  try
+  {
+    return py::cast(self.value<T>(port));
+  }
+  catch(std::runtime_error const& e)
```

Comment:
Should there be a catch for `cast_error` too since that is what pybind11 returns if the casting fails? https://pybind11.readthedocs.io/en/stable/advanced/pycpp/object.html#casting-back-and-forth

Reply:
I dont think so. This catch is meant to catch cases where the data is not found at that port. If the type conversion fails, I think we would still want to propagate that to the user?

Reply:
gotcha thanks for the clarification

Reply:
Yes... I admit I had to check, but still throwing if the conversion fails appears consistent with the C++ API. (Also, in Python if you don't want that, you can just use the method that doesn't require specifying what type you expect to get back.)